packages/app-builder-lib/src/electron/electronMac.ts

```diff
@@ -117,7 +117,14 @@ export async function createMacApp(packager: MacPackager, appOutDir: string, asa
   if (oldHelperBundleId != null) {
     log.warn("build.helper-bundle-id is deprecated, please set as build.mac.helperBundleId")
   }
-  const helperBundleIdentifier = filterCFBundleIdentifier(packager.platformSpecificBuildOptions.helperBundleId || oldHelperBundleId || `${appInfo.macBundleIdentifier}.helper`)
+
+  const cfBundleIdentifier = filterCFBundleIdentifier((isMas ? packager.config.mas?.appId : packager.platformSpecificBuildOptions.appId) || appInfo.macBundleIdentifier)
```

Comment:
This will break previous logic if `isMas = true` & `mas?.appId = null`
```
(isMas ? packager.config.mas?.appId : packager.platformSpecificBuildOptions.appId)
```

I think it needs be
```
const defaultAppId = packager.platformSpecificBuildOptions.appId
const cfBundleIdentifier = filterCFBundleIdentifier((isMas ? packager.config.mas?.appId : defaultAppId) || defaultAppId || appInfo.macBundleIdentifier)

const defaultHelperId = packager.platformSpecificBuildOptions.helperBundleId
const helperBundleIdentifier = filterCFBundleIdentifier(
    (isMas ? packager.config.mas?.helperBundleId : defaultHelperId) || defaultHelperId || oldHelperBundleId || `${cfBundleIdentifier}.helper`
  )
```
I honestly don't like the way to write it, but `defaultAppId` is required twice, unfortunately.

Reply:
Thanks for the heads up @mmaietta ! You are right, I have pushed a commit with your suggestion.